models/Comment.php

```diff
@@ -66,60 +62,57 @@ function get_comments(){
 
     // Fonction permettant de supprimer un commentaire
     function delete_comment(){
-        if (isset($_GET['delete']) AND !empty($_GET['delete'])) {
+        $delete = filter_input(INPUT_GET, 'delete', FILTER_SANITIZE_NUMBER_INT);
         
-        $delete = (int) $_GET['delete'];
-
-        $db = getPdo();
-        
-        $req = $db->prepare('DELETE FROM comments WHERE id = ?');
+        $req = $this->db->prepare('DELETE FROM comments WHERE id = ?');
         $req->execute(array($delete));
 
-        header("Location: index.php?page=dashboard");
-        
-        exit();    
-        }
+        header("Location: /dashboard");
     }
 
     // Fonction permettant d'approuver un commentaire
     function approve_comment(){
-        if (empty($_GET['approve']) || !ctype_digit($_GET['approve'])) {
-            die("Ho ! Fallait préciser le paramètre id en GET !");
-        }
-
-        $id = $_GET['approve'];
+        $id = filter_input(INPUT_GET, 'approve', FILTER_SANITIZE_NUMBER_INT);
 
-        $db = getPdo();
-
-        $query = $db->prepare('UPDATE comments SET seen=1 WHERE id=:id');
+        $query = $this->db->prepare('UPDATE comments SET seen=1 WHERE id=:id');
+        
         $query->execute(['id' => $id]);
 
-        header("Location: index.php?page=dashboard");
-
-        exit();
+        header("Location: /dashboard");
     }
 
-    // EN COURS
-    //Fonction permettant de supprimer tous les commentaires d'un article (dans le cas où on veut supprimer un article)
-    function delete_article_comments(){
-        
-        if (empty($_GET['id']) || !ctype_digit($_GET['id'])) {
-            die("Ho ! Fallait préciser le paramètre id en GET !");
+    
+    // Fonction permettant de vérifier que les données envoyées pat l'utilisateur sont bonnes.
+    function form_comment_verification(){
+        if(filter_has_var(INPUT_POST, 'submit')){
+            $name = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'name', FILTER_SANITIZE_STRING)));
+            $email = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'email', FILTER_SANITIZE_STRING)));
+            $comment = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'comment', FILTER_SANITIZE_STRING)));
+            $errors = [];
+
+            if(empty($name) || empty($email) || empty($comment)){
+                $errors['empty'] = "Tous les champs n'ont pas été remplis";
+            }else{
+                if(!filter_var($email, FILTER_VALIDATE_EMAIL)){
+                    $errors['email'] = "L'adresse email n'est pas valide";
+                }
+            }
+
+            if(!empty($errors)){
+            ?>
+                <div class="card red">
+                    <div class="card-content white-text">
+                        <?php
+                            foreach($errors as $error){
+                                echo $error;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$error'.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5344015)